firebase-firestore/src/main/java/com/google/firebase/firestore/core/CompositeFilter.java

```diff
@@ -0,0 +1,133 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.core;
+
+import android.text.TextUtils;
+import androidx.annotation.Nullable;
+import com.google.firebase.firestore.model.Document;
+import com.google.firebase.firestore.model.FieldPath;
+import com.google.firebase.firestore.util.Function;
+import com.google.firestore.v1.StructuredQuery.CompositeFilter.Operator;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Represents a filter that is the conjunction or disjunction of other filters. */
+public class CompositeFilter extends Filter {
+  private final List<Filter> filters;
+  private final Operator operator;
+
+  public CompositeFilter(List<Filter> filters, Operator operator) {
+    this.filters = filters;
+    this.operator = operator;
+  }
+
+  public List<Filter> getFilters() {
+    return filters;
+  }
+
+  public Operator getOperator() {
+    return operator;
+  }
+
+  @Override
+  public List<FieldFilter> getFlattenedFilters() {
+    List<FieldFilter> result = new ArrayList<>();
```

Comment:
Would it make sense to memoize this result?

Reply:
Please add TODO, otherwise we will never look at this code again.

Reply:
Done